Typescript overhaul

dacebt opened the pull request:
## Overview

This update is a overhual of the codebase. The purpose of this change is to stream line the workflow and create an easier experience onboarding and updating the codebase.

## Changes Overview

Almost every file has been touched or refactored in some way. Some of the highlights include:

- All JSX components converted to typescript
- Deleted all old files and unused components
- Updated Naming of components, functions, methods
- Added Types, interfaces

## Reworked Changes
- Reimagined Context Setup. Updating methods with comments and cleaning up the data flow through the project.
- Reimagined the workflow for sarcophagus containers and decreasing complexity by removing extra components and passing of props.
- The Timer Hook has been revamped and allows for better performance and accuracy. Major Difference: Removed UI being updated by state in hook and allow for component to update UI.
- Archived Tomb now shows the Archived status in the status badge: Ex. If a sarcophagus has been canceled it will now state so in the status badge instead of 'Archived' with the status being shown in a Tooltip. (Tooltip still exists)
- Archaeologist select list (Create Sarcophagus) got a slight performance overall. Better responsiveness when selecting archaeologist. the table resizes cleaner and fits the margin of the container better.
- Using CSS instead of state for selecting archaeologist.
- Added Loader (ScreenShot below)
- Updated Buttons and Links with small action styling to give them some life.
- Added new Badge Labels
- Updated Send File workflow for Archaeologist Service v2.


## Screenshot

Loader Component: Logo has a pulsing animation
![image](https://user-images.githubusercontent.com/38386583/134821250-aebbc86d-d6af-4f4a-ba1f-1c84725ce2a3.png)

## Testing
### Misc
- [x] Start flow with fresh account
- [x] Successfully approve account for transactions
### Tomb
- [x] Tomb Page Loads
- [ ] Successfully rewrap sarcophagus
- [x] Tomb (Resurrection Tab) Able to resurrect (download) file using Private Key
- [x] Tomb (Archived Tab) Able to see archived reason correct for each sarcophagus
- [x] Able to Successfully Cancel Sarcophagus
- [x] Able to Successfully Bury Sarcophagus
- [ ] Able to Successfully Clean Sarcophagus

### Create Sarcophagus
- [x] Each tab is shown as previous inputs are entered
- [x] Error messages shows when appropriate
- [x] Upload Files up to 2.9MB
- [x] When selecting resurrection time it changes the time accurately
- [x] Submitting redirects and shows pending sarcophagus
- [x] Able to Successfully Create Sarcophagus
- [x] Once Created Sarcophagus file is uploaded to Archaeologist Service
- [x] When mining, user is alerted when trying to navigate or close the browser
- [x] When mining is done, file is properly validated and user can sign final transaction

### Resurrect
- [x] Enter Private key displays all the user received sarcophagi, allowing resurrection on those available for download
- [x] File successfully downloaded

### Horus
- [ ] Able to Successfully Accuse Sarcophagus

dacebt committed:
added typescript package and basic types

dacebt committed:
added updated web3 typescript component

dacebt committed:
re added blockchain store

dacebt committed:
created/readded sarcophagi store

dacebt committed:
created Archaeologist store

dacebt committed:
updated store interfaces

dacebt committed:
fixed initial load
- updated with useCallbacks and useEffect for clean initial load

dacebt committed:
refactored Site Header
- removed old header components

dacebt committed:
Readded footer

dacebt committed:
refactor of tomb layout components
- updated components to typescript
- updated some stylings

dacebt committed:
updated transaction hook

dacebt committed:
added contract methods
- Will need continued refactoring for types and optimizing

dacebt committed:
create sarcophagus container
- in progress more refactoring is needed.

dacebt committed:
Created ResurrectionTimer

dacebt committed:
added expand options

dacebt committed:
-> typescript

dacebt committed:
renamed env

dacebt committed:
Embalmer Tomb refactor
- UseCheckStatus and useSendFile need more work
- Recreated Sarcophagus Containers
- updated Contract Method types
- ReCreated expanded sections of Sarcophagus Containers
- Rewrote Fees and Resurrection Forms adding to Shared folder

dacebt committed:
Refactor Tomb Resurrect
- Refactor useRecipientCheckStatus
- Refactor Resurrection component
- added interfaces
- tested download

dacebt committed:
ReFactor Archived Sarcophagi

dacebt committed:
Refactor of Create Sarcophagus Complete
- Refactor how Archaeologists are filtered/selected/disabled
- Change from grid to flexbox for arch table
- cleaned up methods for selecting arch

dacebt committed:
CreateWorkflow Tweeks
- Fixes Submit button not being available without doubleclick of arch
- adds pending sarcophagus to create process
- updates cancelSarcophagus flow to work properly

dacebt committed:
added complete creation workflow

dacebt committed:
General Cleanup
- updated contract methods with types
- added successRefresh method to contract methods
- cleaned styling for archaeologist select
- removed default value for 'timeselect'
- changed ping console to warning

dacebt committed:
readded Public key page

dacebt committed:
Re added Resurrection page
- Reuses new Sarcophagus Container component
- fixed styling a little
- added loader component

dacebt committed:
Readds horus accuse page

dacebt committed:
realigned header

dacebt committed:
updated toast message to dark theme

dacebt committed:
Added Loader to tomb

dacebt committed:
added styling to actions on buttons and links

dacebt committed:
removed remaining javascript files.

sethhrbek commented on src/App.tsx:
```diff
@@ -0,0 +1,20 @@
+        <SiteHeader />
+        <Switch>
+        <Body />
+      </Switch>
```
This indentation looks off for `<Body>` and `</Switch>` -- maybe it's just github?

sethhrbek commented on src/assets/styles/containers.enum.ts:
```diff
@@ -0,0 +1,3 @@
+export enum Container {
+  
+}
```
Why export an empty enum?

dacebt replied:
missed that, will delete

sethhrbek commented on src/components/SarcophagusCreate/ArchaeologistSelect/ArchaeologistData.tsx:
```diff
@@ -0,0 +1,99 @@
+    !archaeologist.accusedCount ||
+    !archaeologist.cleanupCount ||
+    !archaeologist.canceledCount ||
+    !archaeologist.successesCount
```
What is the purpose of checking all these counts here? Is it possible one of them would be null but not the others?

dacebt replied:
I need to go through and check for these typescript conditionals. I need to update how these types are handled

sethhrbek commented on src/components/SarcophagusCreate/ArchaeologistSelect/index.tsx:
```diff
@@ -0,0 +1,62 @@
+};
+
+const ArchaeologistSelect = ({ errors, touched, ...rest }: SelectArchaeologistProps) => {
+  const archaeologistsStore: IArchaeologistsStore = useArchaeologistsStore();
```
Do you need to prefix `I` on this interface? You aren't doing it on other interfaces, and general consensus is interfaces in Typescript shouldn't be prefixed this way.

dacebt replied:
yeah missed that when I went through and updated to not use it. I'll get this updated (and any others)

dacebt replied:
This was fixed in a commit that outdated this comment

sethhrbek commented on src/components/SarcophagusCreate/ArchaeologistSelect/index.tsx:
```diff
@@ -0,0 +1,62 @@
+      <div className="hide-scrollbar overflow-x-scroll w-full whitespace-nowrap">
+        <ErrorText isVisible={!!errors.address && !!touched.address} text={errors.address} />
+        <div className="flex arch-table-row">
+          {archaeologistsTableHeaders(archaeologistsStore.archaeologistsWithStats).map((props: any) => (
```
Why type `any` here? Will these prop types not be static?

dacebt replied:
missed when defining types. I'll check for this in other files as well.

dacebt replied:
This was fixed a commit and is outdated

sethhrbek commented on src/components/SarcophagusCreate/ArchaeologistSelect/index.tsx:
```diff
@@ -0,0 +1,62 @@
+
+const ArchaeologistSelect = ({ errors, touched, ...rest }: SelectArchaeologistProps) => {
+  const archaeologistsStore: IArchaeologistsStore = useArchaeologistsStore();
+  const pagination = usePagination(22);
```
What is `22` here? This should be a constant with a descriptive name of what its purpose is.

dacebt replied:
yeah possibly left in for during testing. I believe this is fixed in another PR, if not I'll update this.

dacebt replied:
this has been updated

sethhrbek commented on src/components/SarcophagusCreate/ArchaeologistSelect/index.tsx:
```diff
@@ -0,0 +1,62 @@
+                {...rest}
+              />
+            ))}
+          {archaeologistsStore.archaeologistsWithStats.length >= 5 && <PageSelect {...pagination} />}
```
`5` should be set as a named constant

sethhrbek commented on src/components/SarcophagusCreate/SelectResurrectionTime/index.tsx:
```diff
@@ -0,0 +1,64 @@
+              If you don’t attest before this time; only the inner layer controlled by the recipient is
+              protecting the data.
+            </div>
+            <div className="pb-4 leading-5">
+              The further you set the resurrection time, the more it will cost.
+            </div>
```
Not a code note, but is this true? Didn't think the system functioned that way.

dacebt committed:
fixed text input not showing

dacebt committed:
adds in toast error for invalid file on download

dacebt committed:
Updated Button Component

dacebt committed:
removed test value

dacebt committed:
clear up:
- rename interfaces
- added a few commits
- cleaned up random spacing and syntax

dacebt committed:
updated env id

dacebt committed:
remove left over console log

dacebt committed:
addes archaeologist endpoint to metrics

dacebt committed:
updated disabled styles for select archaeologist

dacebt committed:
fixed approval and creation bug (#36)

* fixed approval bug

* greater than or equal

* updates approval useEffect

* ensures correct sarcophagus is used in file send

* Allows Error during Arweave to show properly
- removed unused variable

dacebt committed:
removed unnessary custom conditional (#37)

sethhrbek commented on src/components/SarcophagusCreate/hooks/useFileEncryption.ts:
```diff
@@ -1,41 +1,26 @@
-        true
-      ).slice(1);
-      const encrypted = await encrypt(recipPubKeyBytes, fileByteArray);
+      }
```
(Minor) -- will public key length always be the same length? If so, may be better to check the length rather than beginning of the string.

sethhrbek commented on src/components/SarcophagusCreate/hooks/useFileEncryption.ts:
```diff
@@ -1,41 +1,26 @@
-      ).slice(1);
-      const encrypted = await encrypt(recipPubKeyBytes, fileByteArray);
+      }
+      const keyToUse = formatedPublicKey ? formatedPublicKey : recipientPublicKey ? recipientPublicKey : "";
```
misspelling -- `formattedPublicKey`

sethhrbek commented on src/components/SarcophagusCreate/hooks/usePagination.ts:
```diff
@@ -1,17 +1,21 @@
     useEffect(() => { 
         let arr = []
-        for(let i = 0; i < totalPages; i++) arr.push(i)
+        if(Number.isInteger(totalPages)) {
+          for(let i = 0; i <= totalPages - 1; i++) arr.push(i)
+        } else {
+          for(let i = 0; i <= totalPages; i++) arr.push(i)
+        }
```
What does the `else` statement cover here? What would `totalPages` be if not an integer?

dacebt replied:
I agree not sure what thats for. will look into and other I'll remove

dacebt replied:
the problem came with how the number of pages was being calculated. this has been fixed to round the total pages prior to this method.

sethhrbek commented on src/components/SarcophagusCreate/sarcophagusCreate.interfaces.ts:
```diff
@@ -0,0 +1,58 @@
+  daysDisplayed: number;
+  custom: boolean;
+  customTime: string;
+  timeSelect: "week" | "month" | "threeMonths" | "custom" | null;
```
This may be better as a separate enum --- 

```
enum ResurrectionTimeInterval {
  week,
  month,
  threeMonths,
  custom
}
```

`timeSelect: ResurrectionTimeInterval | null`

sethhrbek commented on src/components/SarcophagusCreate/sarcophagusCreate.utils.ts:
```diff
@@ -0,0 +1,77 @@
+      .test("required", "Recipient Address is required", (value) => !!value)
+      .test("validDataHextString", "Please enter a valid public key", (value) => {
+        let testValue;
+        const str = value?.substr?.(0, 4);
```
Why would `value` be null here?

dacebt replied:
I think this is a typescript added value. I need to go through and properly cast the types into components

dacebt replied:
Yup value is passed through as type 'string | undefined' so these are needed.

sethhrbek commented on src/components/SarcophagusCreate/sarcophagusCreate.utils.ts:
```diff
@@ -0,0 +1,77 @@
+    custom: Yup.bool(),
+    fileUploaded: Yup.mixed()
+      .test("required", "Please upload a file", (value) => !!value)
+      .test("filesize", "Files must not be larger than 2.8MB", (value) =>
```
We mention `2.9MB` as the max size elsewhere in this PR.

dacebt replied:
Missed that will update.

sethhrbek commented on src/components/SarcophagusTomb/EmbalmerSarcophagi/index.tsx:
```diff
@@ -0,0 +1,93 @@
+      setStatus={updateStatus}
+      toggleExpansion={() => setIsExpanded((expanded: boolean) => !expanded)}
+      sarcophagus={sarcophagus}
+      isExpandable={!!getExpandsionText(sarcophagusStatus)}
```
misspelling -- `getExpansionText`

sethhrbek commented on src/components/SarcophagusTomb/TombNavigation/index.tsx:
```diff
@@ -0,0 +1,65 @@
+    <div className={Wrapper.TombNavigation}>
+      <NavLink
+        activeClassName="border-b-2 border-white"
+        className="px-2 pb-4 cursor-pointer whitespace-nowrap flex justif-center items-center flex-shrink-0"
```
mispelled classname `justif-center`

sethhrbek commented on src/components/SarcophagusTomb/hooks/useArchaeologistService.tsx:
```diff
@@ -0,0 +1,143 @@
+      if (!fileValid) {
+        if (!tries) {
+          setSendStatus(ServiceStatus.Failed);
+        }
```
Should this also break out of this function? Otherwise `await validateFileSent();` will still get called even if this clause gets hit.

sethhrbek commented on src/components/SarcophagusTomb/hooks/useArchaeologistService.tsx:
```diff
@@ -0,0 +1,143 @@
+    return fileEncoded;
+  };
+
+  const sendFile = async (doubleEncryptedFile?: Uint8Array, endpoint?: string) => {
```
Why are these params optional? If either of these params are null, maybe this method shouldn't be called?

dacebt replied:
most likely a typescript casting issue. I'll update this

dacebt replied:
Ending up solving s few things working through this one.

sethhrbek commented on src/components/SarcophagusTomb/hooks/useArchaeologistService.tsx:
```diff
@@ -0,0 +1,143 @@
+      };
+      const responseFromArch = await fetch(archEndpoint, params);
+      if (!responseFromArch.ok) {
+        if (responseFromArch.status === 406) {
+          return { error: "try again" };
+        }
+        return { error: responseFromArch };
```
What happens in the UI if either of these errors are returned? Is there a difference to the user whether "try again" or a different error is returned?

dacebt replied:
only the error: responseFromArch shows an error to the user. "try again" is purely for running requesting again. I can update this to be a little more clear

dacebt replied:
This process has been updated.

sethhrbek commented on src/components/SarcophagusTomb/hooks/useCheckEmbalmerStatus.ts:
```diff
@@ -0,0 +1,117 @@
+            setSarcophagusStatus(SarcophagusStatus.Signing);
+            return;
+          case ServiceStatus.Mining:
+            if (sarcophagusStatus === SarcophagusStatus.ArweaveMining) return;
```
wouldn't this line only get hit if `sarcophagusStatus === SarcophagusStatus.ArweaveUploading`?

dacebt replied:
Can confirm this is in place to precent too many re-renders when it switch to mining on arweave.

sethhrbek commented on src/components/SarcophagusTomb/hooks/useCheckEmbalmerStatus.ts:
```diff
@@ -0,0 +1,117 @@
+      }
+      // no status is set;
+      default: {
+        if (stopChecking) return;
+        // if sarcophagus is active
+        if (sarcophagus?.assetId && sarcophagus?.privateKey === PRIVATE_KEY_DEFAULT) {
+          setSarcophagusStatus(SarcophagusStatus.Active);
+          return;
+        }
+        // if sarcophagus creation is in process
+        const isCreatedSarcophagus =
+          createdSarcophagusData?.assetDoubleHash ||
+          new Uint8Array() === Buffer.from(ethers.utils.arrayify(sarcophagus.AssetDoubleHash));
+        if (createdSarcophagusData && !sendStatus && isCreatedSarcophagus) {
+          setSarcophagusStatus(SarcophagusStatus.ArweaveUploading);
+          setSendStatus(ServiceStatus.Sending);
+          sendFileToArchService();
+          return;
+        }
+        if (sarcophagusStatus === SarcophagusStatus.Default) {
+          setSarcophagusStatus(SarcophagusStatus.Error);
+          setStopChecking(true);
+        }
```
This block of code is a little confusing -- either breaking these pieces of code into named functions or adding more explicit comments may help define what each of these different cases is for and does.

dacebt replied:
I moved this code into useArchaeologistService hook.

sethhrbek commented on src/components/SarcophagusTomb/hooks/useCheckRecipientStatus.ts:
```diff
@@ -0,0 +1,56 @@
+    );
+    if (isUnwrapped) {
+      setSarcophagusStatus(SarcophagusStatus.Unwrapped);
+    } else if (sarcophagus?.assetId && isActive) {
+      setSarcophagusStatus(SarcophagusStatus.Active);
+    } else if (!sarcophagus?.assetId && isActive) {
+      setSarcophagusStatus(SarcophagusStatus.Created);
```
these could probably be combined into
```
else if (isActive) {
 const sarcoStatus = sarcophagus?.assetId ? SarcophagusStatus.Active : SarcophagusStatus.Created
  setSarcophagusStatus(sarcoStatus)
}
```

dacebt replied:
updated

sethhrbek reviewed (changes requested):
Still working my way through this, but wanted to submit the comments I had so far (all of them are minor or clarification questions).

dacebt committed:
fixes indention

dacebt committed:
delete unused file

dacebt committed:
removes typescript conditional

dacebt committed:
moved number to named constant

dacebt committed:
fixed typo

dacebt committed:
removes unneeded conditional
- finds the number of pages by rounding to next number.
- removes conditional for statement

dacebt committed:
updated Resurrection form time select to enum

dacebt committed:
updated file size to correct amount

dacebt committed:
fixed typo getExpandsionText

dacebt committed:
fixed typo in classname justif-center

dacebt committed:
returns failed when tries have been exhausted

dacebt committed:
updates and optimizes arweave workflow
- added a useEffect and moved start service to custom hook.
- updated service methods, and wrapped in useCallback
- These changes prevent file from being sent to service twice.
- These changes removes optional params that were being passed

dacebt committed:
updated conditional for readibility

dacebt committed:
updates toast message to dark style

dacebt committed:
removes timer while mining

sethhrbek committed:
Merge pull request #38 from sarcophagus-org/comments/typescript-overhaul-01

Comments/typescript overhaul 01

sethhrbek commented on src/components/SarcophagusTomb/hooks/useResurrectionTimers.ts:
```diff
@@ -0,0 +1,61 @@
+    sarcophagus.state !== 2 && sarcophagus.resurrectionWindow ? resurrectionWindow.toNumber() : 0;
+  // calculates total time until resurrection is complete in milli seconds
+  const TimePlusWindowUTCMilli = (ResurrectionTimeUTCSeconds + ResurrectionWindowUTCSeconds) * 1000;
+  const isPastWindow = TimePlusWindowUTCMilli - Date.now().valueOf() <= 0;
+  const isWithinWindow = TimePlusWindowUTCMilli - Date.now().valueOf() <= 0;
```
What is the difference between these constants? They look like they are set to the same value.

dacebt replied:
Yeah I think these got left unchanged after and updated. Will update

sethhrbek commented on src/components/SarcophagusTomb/shared/ResurrectionTimer.tsx:
```diff
@@ -0,0 +1,34 @@
+  if (isStateTwo || !sarcophagus.resurrectionTime) return null;
+  // decides text color of timer
+  const textColor = getTextColor(resurrectionTimerState.timerStatus);
+  if (resurrectionTimerState.timerStatus === TimerStatus.Off) return null;
```
Style note -- this is just personal preference, but prefer grouping early returns together and as early in the function call as possible. This way, when reading the function its quickly clear which cases cause the function to return.

sethhrbek commented on src/components/SarcophagusTomb/shared/Rewrap/index.tsx:
```diff
@@ -0,0 +1,184 @@
+  }, [approved]);
+
+  const handleApproval = (errors: RewrapFormErrors) => {
+    if (!!Object.keys(errors).length) return;
```
What happens in the case of an error here? This doesn't throw an error, so does this get handled somewhere?

sethhrbek commented on src/components/SarcophagusTomb/shared/Rewrap/index.tsx:
```diff
@@ -0,0 +1,184 @@
+
+  const initialValues: RewrapFormState = {
+    resurrectionTime: getDateInFuture(7),
+    bounty: getDecimalNumber(archaeologist?.minimumBounty || ethers.BigNumber.from(0), 18) || 0,
+    diggingFee: getDecimalNumber(archaeologist?.minimumDiggingFee || ethers.BigNumber.from(0), 18) || 0,
```
If archaeologist is null here, will this form work? Trying to determine if we could reasonably expect archaeologist to be null here.

sethhrbek commented on src/components/SarcophagusTomb/shared/Rewrap/index.tsx:
```diff
@@ -0,0 +1,184 @@
+              onClick={burySarcophagus}
+            >
+              <span className="mr-2 cursor-pointer">Bury sarcophagus</span>
+              <Tooltip content="Burying a sarcophagus, releases digging fees to archaeologist and archives sarcophagus." />
```
can remove the comma here

sethhrbek commented on src/components/SarcophagusTomb/shared/Rewrap/index.tsx:
```diff
@@ -0,0 +1,184 @@
+                      validateForm();
+                      handleApproval(errors);
+                    }
+              }
```
Should we validate the form in the case where approval has already happened? Could the user update the form values after approval leading to an invalid form submission?

sethhrbek commented on src/components/SarcophagusTomb/shared/SarcophagusExpandedSection.tsx:
```diff
@@ -0,0 +1,133 @@
+const SectionContainer = ({ text, tooltipContent, action, signButton }: SectionContainerProps) => {
+  return (
+    <div className="flex flex-col items-center justify-center h-full gap-8" style={{ height: "12.0625rem" }}>
+      {signButton && signButton}
```
Would just `signButton` work here?

sethhrbek commented on src/components/SarcophagusTomb/shared/SarcophagusExpandedSection.tsx:
```diff
@@ -0,0 +1,133 @@
+  if (!isExpanded) return null;
+
+  switch (status) {
+    case SarcophagusStatus.Accused:
+    case SarcophagusStatus.ArweaveMining:
+    case SarcophagusStatus.ArweaveUploading:
+    case SarcophagusStatus.ArchivedUnwrapped:
+    case SarcophagusStatus.Mining:
+    case SarcophagusStatus.Created:
+    case SarcophagusStatus.Buried:
+    case SarcophagusStatus.Canceled:
+    case SarcophagusStatus.Cleaned:
+    case SarcophagusStatus.Default:
+    case SarcophagusStatus.Unwrapping:
+    case SarcophagusStatus.Archived:
```
Could `return null` be moved to the `default` case in the switch statement, and then you don't need to check for all these states?

dacebt committed:
fixes rewrap bug
- Rewrap functionality was broken
- updated to match type of data that is passed
- updated timer to recalculate time after rewrap

dacebt committed:
updates unwrap to happen automatically

dacebt committed:
reloads archaeologists on action execute

sethhrbek commented on src/components/SarcophagusTomb/shared/SarcophagusContainer.tsx:
```diff
@@ -0,0 +1,95 @@
+        <div className="flex flex-col">
+          <ExpandButton isExpandable={isExpandable} isExpanded={isExpanded} status={status} />
+
+          <SarcophagusStatusBadge status={status} isActive={status === SarcophagusStatus.Active} isArchived={sarcophagus.state === 2}/>
```
For the future (doesn't need to be handled in this PR) -- should have an enum for these different sarco states (0/1/2)

sethhrbek commented on src/components/SarcophagusTomb/shared/SarcophagusExpandedSection.tsx:
```diff
@@ -0,0 +1,133 @@
+    sarcophagiStore.updateSarcophagus(setStatus, successRefresh);
+  };
+
+  if (!isExpanded) return null;
```
Can this be at the top of this component?

sethhrbek commented on src/components/SarcophagusTomb/tomb.utils.ts:
```diff
@@ -0,0 +1,100 @@
+  });
+};
+
+export const arweaveFileValid = async (
+  arweave: Arweave,
+  transactionId?: string,
+  doubleEncryptedFile?: string | Uint8Array
+) => {
+  if (!transactionId || !doubleEncryptedFile) {
+    return false;
+  }
+  try {
+    const data = await arweave.transactions.getData(transactionId, { decode: true });
+    const buffedData = Buffer.from(data);
+    const buffFile = Buffer.from(doubleEncryptedFile);
+    return Buffer.compare(buffedData, buffFile) === 0;
+  } catch (e) {
+    console.error(e);
+  }
+};
```
(minor) -- include Typscript return type (boolean)

sethhrbek commented on src/components/SarcophagusTomb/tomb.utils.ts:
```diff
@@ -0,0 +1,100 @@
+    const buffFile = Buffer.from(doubleEncryptedFile);
+    return Buffer.compare(buffedData, buffFile) === 0;
+  } catch (e) {
+    console.error(e);
```
Should this return `false`, since file is not valid if error is thrown?

sethhrbek commented on src/components/SarcophagusTomb/tomb.utils.ts:
```diff
@@ -0,0 +1,100 @@
+    case SarcophagusStatus.Error:
+    case SarcophagusStatus.ArweaveMiningError:
+      return "Details";
+    default:
```
Will this default ever be hit (where it returns an empty string)? If not, you could make this return "Details" and remove the 4 cases above.

sethhrbek commented on src/components/SarcophagusTomb/tomb.utils.ts:
```diff
@@ -0,0 +1,100 @@
+  }
+};
+
+export const hexString = (value: string) => {
+  let hexKey;
+  if (value?.substr(0, 2) !== "0x") hexKey = "0x" + value;
+  return hexKey || value;
+};
```
This could be shortened to:
`return (value.substr(0, 2) !== "0x") ? "0x" + value : value`

sethhrbek commented on src/components/shared/ResurrectionForm/ResurrectionTimeForm/index.tsx:
```diff
@@ -0,0 +1,113 @@
+
+const ResurrectionTimeForm = ({ values, handleChange, setFieldValue }: ResurrectionTimeFormProps) => {
+  const weekAhead = getDateInFuture(7);
+  const monthAhead = getDateInFuture(30);
+  const threeMonthAhead = getDateInFuture(90);
```
(minor, for the future) Don't update in this PR -- would be nice to calculate exactly 1 or 3 months ahead, as # of days in month can vary.

sethhrbek commented on src/components/shared/ResurrectionForm/index.tsx:
```diff
@@ -0,0 +1,116 @@
+        Buffer.from(doubleEncryptedData)
+      );
+
+      // decrypt with public key
```
(minor) -- this recipients private key

sethhrbek commented on src/components/shared/components.utils.ts:
```diff
@@ -0,0 +1,116 @@
+  return makeNumeral(bigNumber, decimals).value();
+};
+
+export const getNumberalString = (bigNumber: BigNumber, decimals: number, isFixed?: boolean) => {
```
Misspelling on this function name

sethhrbek commented on src/components/shared/components.utils.ts:
```diff
@@ -0,0 +1,116 @@
+  return makeNumeral(bigNumber, decimals).input();
+};
+
+export const archTotalFees = (archaeologist: Archaeologist, file: File | null, returnBigNumber?: boolean) => {
```
Is `returnBigNumber` ever passed into this function? I searched around and it didn't look like it was being used (but may just not be in the diff of this PR).

sethhrbek commented on src/components/shared/components.utils.ts:
```diff
@@ -0,0 +1,116 @@
+export const getStorageFee = (archaeologist: Archaeologist, file: File | null, returnBigNumber?: boolean) => {
+  const { feePerByte } = archaeologist;
+  if (!feePerByte) return "";
+  if (!file?.size) return "";
```
This could be the first line in this function

sethhrbek commented on src/containers/PublicKey/index.tsx:
```diff
@@ -0,0 +1,81 @@
+      const msg = "Hello from the Sarcophagus! Sign this message to retrieve your account's public key";
+      const msgHash = utils.hashMessage(msg);
+      const msgHashBytes = utils.arrayify(msgHash);
+      const signature = await signer?.signMessage(
```
I think this null check on `signer` should be removed, and if the signer doesn't exist an error will be thrown.

sethhrbek commented on src/containers/SarcophagusCreate/index.tsx:
```diff
@@ -0,0 +1,190 @@
+  };
+
+  const createSarcophagus = async (values: SarcophagusCreateValues) => {
+    if (!selectedArchaeologist || !storageFee || !assetDoubleHash || !doubleEncryptedFile) {
+      console.error("🚀 ~ Create values not found");
+      return;
+    }
```
It may help for debugging purposes to be more explicit in the logs about which one of these is null. Could log all of the vars out here if you don't want to change the conditional logic.

Also, what happens if this returns? Does the UX notify the user they have to take an action?

sethhrbek commented on src/containers/SarcophagusCreate/index.tsx:
```diff
@@ -0,0 +1,190 @@
+      const diggingFeeBN = utils.parseEther(diggingFee.toString());
+      const bountyBN = utils.parseEther(bounty.toString());
+      const storageFeeBN = utils.parseEther(storageFee.toString());
+      let formatedPublicKey;
+      if (recipientPublicKey.substr(0, 4) !== "0x04") formatedPublicKey = "0x04" + recipientPublicKey;
```
This`0x04` prefix is handled elsewhere in the same manner. Could refactor this into a utils formatter function so you could just write:

```
const recipientPublicKeyBA = utils.arrayify(formatPublicKey(recipientPublicKey)).slice(1);
```

sethhrbek commented on src/containers/SarcophagusResurrection/index.tsx:
```diff
@@ -0,0 +1,138 @@
+        <PrivateKeyField handleQuery={handleQuery} initialValues={initialValues} />
+      </div>
+      <div className="" style={{ width: "34rem" }}>
+        {!isRecipientSarcophagiLoaded && recipientPrivateKey ? (
```
What happens if `isRecipientSarcophagiLoaded` is `true`, but `recipientPrivateKey` is null? Should this be:
`!isRecipientSarcophagiLoaded || !recipientPrivateKey`?

sethhrbek replied:
Or do we want to display an error message if the `recipientPrivateKey` is null, b/c that means the query value in URL wasn't set?

sethhrbek reviewed (changes requested):
Left some comments -- all minor stuff, overall looks great!

sethhrbek committed:
Merge pull request #39 from sarcophagus-org/typescript/update-toast-to-dark

updates toast message to dark style

sethhrbek committed:
Merge pull request #42 from sarcophagus-org/typescript/bugfix/arch-public-key-refresh

reloads archaeologists on action execute

sethhrbek committed:
Merge pull request #40 from sarcophagus-org/typescript/rewrap-bug-fix

Typescript/rewrap bug fix

sethhrbek committed:
Merge pull request #35 from sarcophagus-org/check-for-max-resurrection-time

updated disabled styles for select archaeologist

sethhrbek committed:
Merge pull request #34 from sarcophagus-org/metrics-endpoint

addes archaeologist endpoint to metrics

dacebt committed:
returns after conditional properly, fixes throwing error too soon

dacebt committed:
deselects archaeologist (#32)

- removes a console log
- fixed some styling

dacebt committed:
Cleanup/types (#44)

* moved create exported interface to /types folder

* moved tomb interfaces and enum to /types

dacebt committed:
aligned expand text to right (#45)
